promote-release-to-box.py

```diff
@@ -0,0 +1,290 @@
+    for build_artifact in tqdm(builds_output_json):
+        file_name = build_artifact['path']
+        if build_artifact['path'].split('/')[-1] in artifacts_to_release.keys():
+
+            logging.info(f"Downloading {file_name}")
+
+            path = build_artifact['path']
+            sha1 = build_artifact['sha1']
+            sha256 = build_artifact['sha256']
+            md5 = build_artifact['md5']
+
+            file_name = build_artifact['path'].split('/')[-1]
+            check_sum_file_name = path.split('/')[-1]+'.sum'
+
+            # download artifact
+            rt_path = ArtifactoryPath(
+                f"{rt_base_url}/"+path, auth=auth, auth_type=HTTPBasicAuth
+            )
+            
+            if not os.path.exists(file_name):
+                logging.info(f'Downloading {file_name} from {rt_path}.')
+                rt_path.writeto(out=file_name, progress_func=None)
+                logging.info(f'Download complete.')
+            
+            if not os.path.exists(check_sum_file_name):
+                logging.info(f'Saving check sum file {check_sum_file_name}')
+                with open(check_sum_file_name, 'w') as f:
+                    f.write('sha1='+ sha1 +'\n')
+                    f.write('sha256='+ sha256 +'\n')
+                    f.write('md5='+ md5+'\n')
+                
+        release_artifact_downloaded.append(file_name)
+                
+    logging.info(f'Release artifacts {release_artifact_downloaded}')
+    return release_artifact_downloaded
+
+
+def replace_versions(text, replacement_word):
+    pattern = r'\${.*?}'
+    return re.sub(pattern, replacement_word, text)
+
+
+def get_manifest_yaml(version, manifest_file = 'manifest.yaml'):
+    with open(manifest_file, 'r') as f:
+        file_content = f.read()
+
+    new_file_content = replace_versions(file_content, version)
+    yaml_obj = yaml.safe_load(new_file_content)
+    return yaml_obj
+
+
+def process_manifest_yaml(yaml_data, parent=None):
+    '''
+    Returns a dict with parent as value, child as key.
+    {'pad-ee-9.5.1.0-dist.zip': 'ee/client-tools',
+     'pdi-ee-client-9.5.1.0-dist.zip': 'ee/client-tools',
+     'pentaho-analysis-ee-9.5.1.0-dist.zip': 'ee/client-tools',
+     'pentaho-big-data-ee-package-9.5.1.0-dist.zip': 'ee/client-tools',
+     'pme-ee-9.5.1.0-dist.zip': 'ee/client-tools',
+     'prd-ee-9.5.1.0-dist.zip': 'ee/client-tools',
+     ...}
+    
+    '''
+    result = {}
+    for key, value in yaml_data.items():
+        current_key = key if parent is None else f"{parent}/{key}"
+        if isinstance(value, dict):
+            result.update(process_manifest_yaml(value, parent=current_key))
+        elif isinstance(value, list):
+            for item in value:
+                if '$' not in item or '{' not in item:
+                    result[item] = current_key
+    logging.info(f'Read manifest {result}')
+    return result
+
+
+def get_manifest_buildinfo_intersect(file_folder_dict, builds_output_json):
```

Comment:
The function could have some comments also.

Reply:
added comments